src/request-pipeline/context.js

```diff
@@ -126,7 +126,8 @@ export default class RequestPipelineContext {
             if (parsedReferer) {
                 this.dest.referer   = parsedReferer.dest.url;
                 this.dest.reqOrigin = urlUtils.getDomain(parsedReferer.dest);
-            }
+            } else if (this.req.headers[XHR_HEADERS.origin])
```

Comment:
Please, place `else if` on the newline. Odd, I thought we've had eslint rule for this